projects/iio_demo/src/app/main.c

```diff
@@ -194,12 +212,48 @@ int main(void)
 		.size = MAX_SIZE_BASE_ADDR
 	};
 
+	struct iio_data_buffer adc_buff = {
+		.buff = (void *)ADC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR
+	};
+
+	struct iio_data_buffer dac_buff = {
+		.buff = (void *)DAC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR
+	};
+
+	adc_init_par = (struct adc_demo_init_param) {
+		.loopback_buffers = NULL,
+		.channel_no = 16,
+		.dev_global_attr = 3333,
+		.dev_ch_attr = {1111,1112,1113,1114,1115,1116,1117,1118,1119,1120,1121,1122,1123,1124,1125,1126}
+	};
+	status = adc_demo_init(&adc_desc, &adc_init_par);
+	if (status != SUCCESS)
+		return status;
+
+	dac_init_par = (struct dac_demo_init_param) {
+		.loopback_buffers = NULL,
+		.channel_no = 2,
+		.dev_global_attr = 4444,
+		.dev_ch_attr = {1111,1112,1113,1114,1115,1116,1117,1118,1119,1120,1121,1122,1123,1124,1125,1126}
+	};
+	status = dac_demo_init(&dac_desc, &dac_init_par);
+	if (status != SUCCESS)
+		return status;
+	//MODIFIED_CHANNEL_VALUE = 16;
+	uint32_t adc_channels = ADC_CHANNEL_NO;
+	struct iio_device adc_demo_iio_descriptor = ADC_DEMO_DEV(adc_channels);
+
+	uint32_t dac_channels = DAC_CHANNEL_NO;
+	struct iio_device dac_demo_iio_descriptor = DAC_DEMO_DEV(dac_channels);
+
 	struct iio_app_device devices[] = {
-		IIO_APP_DEVICE("demo_device_output", iio_demo_out_desc,
```

Comment:
When removing this you also have to remove remaining old iio_demo structures.
Anyway, maybe you can just keep old ones in this commit and in an other commit remove this an all related with old iio_demo